File: src/infrastructure/tcp/servers/gate/gate-server.ts
import { BaseTcpServer } from "../../../../core/tcp/services/base-tcp-server-service.ts";
import "./commands/get-news.ts";
import "./commands/get-lobby-list.ts";

export class GateServer extends BaseTcpServer {
  protected readonly serverType = "gate" as const;
  protected readonly port: number;

  constructor(port: number) {
    super();
    this.port = port;
  }
}

File: src/GateLobbyServer/GateServer.cs
using Mgo2Server.Shared.Tcp;
using Mgo2Server.Shared.Types;

namespace Mgo2Server.GateLobbyServer;

/// <summary>
/// First connection point of the game: answers the lobby list and the news. It
/// serves the port of the lobby row of type gate, which the server publishes
/// from the environment when it starts.
/// </summary>
public sealed class GateServer(IServiceProvider serviceProvider, int port)
    : TcpServerBase(serviceProvider, port)
{
    /// <inheritdoc />
    protected override ServerType ServerType => ServerType.Gate;

    /// <inheritdoc />
    protected override string LogPrefix => "tcp:gate";
}
